### PR TITLE
Add athena plugin 

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Add athena plugin

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: plugins/athena/flytekitplugins/athena/task.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            inputs=inputs,
+            outputs=outputs,
+            task_type=self._TASK_TYPE,
+            **kwargs,
+        )
+        self._output_schema_type = output_schema_type
+
+    def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
+        # This task is executed using the presto handler in the backend.
+        job = PrestoQuery(statement=self.query_template)
```
Comment (EngHabu) [inline plugins/athena/flytekitplugins/athena/task.py:67]: I stand corrected, I think it needs somethings set... they are just named differently... I think the minimum is the Database. We used [Schema here](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/plugins/webapi/athena/utils.go#L129-L149) to map it to...

Comment (katrogan) [reply plugins/athena/flytekitplugins/athena/task.py:67]: yep you're right! ty

**Comment by katrogan** [comment]
PTAL @kumare3 @EngHabu @wild-endeavor tested with https://github.com/flyteorg/flytesnacks/pull/260

**Comment by wild-endeavor** [comment]
Let's change the top level folder name from athena to something else.

Because this exists https://pypi.org/project/athena/

So `plugins/somethingelse/flytekitplugins/athena/`
otherwise weird things happen.

**Comment by katrogan** [comment]
@wild-endeavor thanks! done

**Review by wild-endeavor — COMMENTED** [review]
either way doesn't matter.

**Comment by wild-endeavor** [comment]
@kumare3 did you still have concerns about this PR?  maybe we should add a test for a dynamic task wrapping an athena task?

File: plugins/flytekit-athena/flytekitplugins/athena/task.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            inputs=inputs,
+            outputs=outputs,
+            task_type=self._TASK_TYPE,
+            **kwargs,
+        )
+        self._output_schema_type = output_schema_type
+
+    def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
+        # This task is executed using the presto handler in the backend.
+        job = PrestoQuery(statement=self.query_template, schema=self.task_config.database)
```
Comment (EngHabu) [inline plugins/flytekit-athena/flytekitplugins/athena/task.py:67]: Do you not want to fill/expose workgroup and catalog too?

Comment (katrogan) [reply plugins/flytekit-athena/flytekitplugins/athena/task.py]: @EngHabu done

**Review by EngHabu — APPROVED** [review]

File: plugins/flytekit-athena/flytekitplugins/athena/task.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            inputs=inputs,
+            outputs=outputs,
+            task_type=self._TASK_TYPE,
+            **kwargs,
+        )
+        self._output_schema_type = output_schema_type
+
+    def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
+        # This task is executed using the presto handler in the backend.
+        job = PrestoQuery(
```
Comment (kumare3) [inline plugins/flytekit-athena/flytekitplugins/athena/task.py:72]: My concern is these settings @wild-endeavor. These are not standard, but I guess ok

**Pull request merged**